src/components/ContentOpenWith/ContentOpenWith.js

```diff
@@ -229,21 +237,23 @@ class ContentOpenWith extends PureComponent<Props, State> {
                 this.executeIntegrationSuccessHandler,
                 this.executeIntegrationErrorHandler,
             );
+
+        this.executeId = appIntegrationId;
     };
 
     /**
      * Opens the integration in a new tab based on the API data
      *
      * @private
-     * @param {ExecuteAPI} executePostData - API response on how to open an executed integration
+     * @param {ExecuteAPI} executeData - API response on how to open an executed integration
 
      * @return {void}
      */
-    executeIntegrationSuccessHandler = (executePostData: ExecuteAPI): void => {
-        const { method, url } = executePostData;
+    executeIntegrationSuccessHandler = (executeData: ExecuteAPI): void => {
+        const { method, url } = executeData;
         switch (method) {
             case HTTP_POST:
-                this.setState({ executePostData });
+                this.setState({ executePostData: executeData });
                 break;
             case HTTP_GET:
```

Comment:
Consider using `setState` for the GET case as well, which would allow the render method to handle the different integration types. You could then move the `onExecute` invocation to the `setState` "post-render" callback.